src/layouts/base.js

```diff
@@ -183,6 +187,11 @@ const BaseLayout = ({
               </StyledLinkWrapper>
               <DimHeaderElement
                 onClick={() => {
+                  if (
+                    !hasInfMkrApproval &&
+                    onboardingState === OnboardingStates.SETUP_LINKED_WALLET
+                  )
+                    return modalOpen(Lock);
```

Comment:
Opening the modal here will prompt the user to finish granting approvals. After that, they will be back in a normal "finished" state. @levity @jparklev What do you think?

Reply:
do you think there's value in refactoring the paths the expect a hot wallet `type` in the onboarding to handle `null` or `undefined`?  if not this approach makes sense to me. one thought that just came up: if we go with this, do we need to ensure they're using their cold wallet?

Reply:
`type` here is e.g. Metamask? If we have almost all the data we need to get through the onboarding process, maybe we can do without `type` as well--I would look into @jparklev's idea. @abylaw, maybe you could chime in here

Reply:
@jparklev digging a little deeper into your idea, it looks like `type` is only used for UI purposes (to display text or an icon). I like this because it involves the least amount of change to business logic.

If we handle null or undefined here, we would need a default value to display in places like this, for example: `{`To proceed with setting up your voting contract, please sign the transaction in ${nicelyFormatWalletProvider(coldWallet.type)}.`}`. If we don't have a wallet type here, how do you feel about generic `your wallet`?

`WalletIcon.js` is the other consumer of `.type` here, but we now have a generic wallet icon, so I would use that as the default case.

I'll move forward with this idea, but just wanted to answer this question:
> if we go with this, do we need to ensure they're using their cold wallet?

The bug only occurs if the user is already connected to their cold wallet, and they don't have approvals in place. If they're using a hot wallet, there are checks in place to guide them back to the right onboarding step.